optimade/validator/validator.py

```diff
@@ -1213,8 +1213,11 @@ def _get_endpoint(
         if response.status_code not in expected_status_code:
             message = f"Request to '{request_str}' returned HTTP code {response.status_code} and not expected {expected_status_code}."
             message += "\nError(s):"
-            for error in response.json().get("errors", []):
-                message += f'\n  {error.get("title", "N/A")}: {error.get("detail", "N/A")} ({error.get("source", {}).get("pointer", "N/A")})'
+            try:
+                for error in response.json().get("errors", []):
+                    message += f'\n  {error.get("title", "N/A")}: {error.get("detail", "N/A")} ({error.get("source", {}).get("pointer", "N/A")})'
+            except json.JSONDecodeError:
+                pass
```

Comment:
We could consider adding some information about the headers, e.g., what the returned media type is?
```suggestion
                message += f'\n  Error response (not JSON-serializable). Return code: {response.status_code}'
```

Reply:
Done; ignored your suggestion itself as the return code is always printed